dwdatareader/__init__.py

```diff
@@ -96,6 +97,41 @@ def description(self):
     def number_of_samples(self):
         return DLL.DWGetScaledSamplesCount(self.index)
 
+    @staticmethod
+    def _c_int():
+        return ctypes.c_int(ctypes.sizeof(ctypes.c_int))
+
+    def _chan_prop_contents(self, chan_prop):
+        buff = ctypes.create_string_buffer(self._c_int().value)
+        p_buff = ctypes.cast(buff, ctypes.POINTER(ctypes.c_void_p))
+        DLL.DWGetChannelProps(
+            self.index, ctypes.c_int(chan_prop.value), p_buff,
+            ctypes.byref(self._c_int()))
+        return ctypes.cast(p_buff,
+                           ctypes.POINTER(ctypes.c_int)).contents
+
+    def _chan_prop_str(self, chan_prop, chan_prop_len):
+        len_str = self._chan_prop_contents(chan_prop_len)
+        p_buff = ctypes.create_string_buffer(len_str.value)
+        DLL.DWGetChannelProps(
```

Comment:
Calls to the DLL should check the return value and raise DWError if necessary.
Also: the p_buff.value.decode() should refer to the module's encoding property.

Reply:
Dito.